### DISCUSSION TITLE
React in tauri

### DISCUSSION THREAD

**Discussion by rdwnsjjd** [tauri-apps/tauri#2911] [General]
Hi there!

Can we use reactjs inside tauri?
or do they interfere each other?

**Comment by FabianLars** [comment]
Yes, tauri doesn't really care about your frontend (as long as it's not SSR).
Also, [create-tauri-app](https://github.com/tauri-apps/tauri/tree/next/tooling/create-tauri-app) provides starter projects with react, either via `create-react-app` or vitejs+react.

**Reply by rdwnsjjd** [reply]
Ok
so how can I create a tauri app that has react lib inside?
am I still should use `npm create tauri-app` command and so on?

**Reply by FabianLars** [reply]
If the react app doesn't exist yet `npm create tauri-app` is the way to go imo. If your react app is already existing (for example you created one with CRA) then there is the `tauri init` command like explained in the [docs](https://tauri.studio/en/docs/usage/development/integration#1-add-tauri-to-existing-project).
In general tauri apps are pretty decoupled from your frontend, so you can set up your react app the same as you would for a browser project, at least until you use tauri's api which isn't browser compatible.

**Reply by rdwnsjjd** [reply]
tnx!

**Reply by unsecretised** [reply]
Wait so does this work if lets say I were to have an existing tauri project and I wanted to add react to it?

**Reply by FabianLars** [reply]
sure, wht not. you basically just swap (or modify) your frontend. All tauri needs to know about your frontend ist the devUrl and frontendDist path
